packages/create-redwood-app/src/create-redwood-app.js

```diff
@@ -67,6 +67,33 @@ export const CreateNewApp = ({ args }) => {
     setMessages(latestMessages.current)
   }
 
+  // set index.html title based on user input directory
+  const changeHtmlTitle = async (path) => {
+    const newTitle = (userDir) => {
+      return userDir
+        .split("/")
+        .slice(-1)[0]
+        .split(/ |-|_/)
+        .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+        .join(' ')
+    }
+
+    fs.readFile(path + '/web/src/index.html', 'utf8', (e, data) => {
+      if (e) {
+        return setNewMessage(e)
+      }
+      var result = data.replace(/\<\s*title[^>]*>(.*?)\<\s*\/\s*title\s*\>/g, '<title>' + newTitle(path) + '</title>')
```

Comment:
Our style is to never use `var`. We use `const` for values that won't change, and `let` for those that will. In general, `const` and writing immutable code is preferred. See https://hackernoon.com/why-you-shouldnt-use-var-anymore-f109a58b9b70

This line is pretty long, it would be nice to split it into several nicely named `const` declarations and pass those in.

Since we control what the HTML in `create-react-app` is, the regex probably doesn't need to be so complex (`\s*` and such affordances), nor does it really need the `g` modifier, as I can't see any reason we would ever put more than one `<title>` element in the page.

I would be tempted to use the character class `/[ _-]/` to split the dir name instead of alternation, it feels more appropriate for single characters.

Reply:
Thanks for all the comments + direction! 
- Article on `var` is great. And happy to stick to `let` and `const`. Sanity = ✔️ 
- Regex is my achilles, so I tend to pull out the sledgehammer. Good catch on unnecessary global. Simplified to `/\<title\>(.*?)\<\/title\>/` Could even go further but then it’s unclear to reader what’s being targeted. 
- cleaned up readability/length with additional `const`. ✔️ 
- Switched to character class suggestion. ✔️